File: usaco/train/holstein/holstein.java
/*
ID: tylan071
LANG: JAVA
TASK: holstein
*/

import java.io.*;
import java.util.*;

class holstein {
    public static int v;
    public static int[] vit;
    public static int g;
    public static int[][] feeds;
    public static ArrayList<Integer> ans = new ArrayList<Integer>();

    public static void main(String[] args) throws IOException {
        read();
        solve();
        write();
    }

    public static void solve() {
        for (int k = 0; k < (int) Math.pow(2, g); k++) {
            int[] amts = new int[v];
            ArrayList<Integer> feedsUsed = new ArrayList<Integer>();
            int set = k;
            for (int i = 0; i < g; i++) {
                if (set % 2 == 1) {
                    feedsUsed.add(i+1);
                    for (int j = 0; j < v; j++) {
                        amts[j] += feeds[i][j];
                    }
                }
                set /= 2;
            }

            boolean isValid = true;
            for (int i = 0; i < v; i++) {
                if (amts[i] < vit[i]) {
                    isValid = false;
                    break;
                }
            }

            if (isValid && feedsUsed.size() < ans.size()) {
                ans = feedsUsed;
            }
        }
    }

    public static void read() throws IOException {
        Scanner in = new Scanner(new File("holstein.in"));

        v = in.nextInt();
        vit = new int[v];
        for (int i = 0; i < v; i++) {
            vit[i] = in.nextInt();
        }

        g = in.nextInt();
        feeds = new int[g][v];
        for (int i = 0; i < g; i++) {
            for (int j = 0; j < v; j++) {
                feeds[i][j] = in.nextInt();
            }
        }

        for (int i = 0; i <= g; i++) ans.add(0);

        in.close();
    }

    public static void write() throws IOException {
        PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("holstein.out")));
        out.print(ans.size());
        for (int i = 0; i < ans.size(); i++) {
            out.print(" " + ans.get(i));
        }
        out.println();
        out.close();
    }
}

File: usaco/train/holstein/holstein.py
"""
ID: tylan071
LANG: PYTHON3
TASK: holstein
"""

with open('holstein.in') as f:
    v = int(f.readline())
    vit = list(map(int, f.readline().split()))
    g = int(f.readline())
    feed = [list(map(int, f.readline().split())) for _ in range(g)]

lo = g + 1
loScoops = 0
for k in range(1, 2**g):
    amts = [0]*v
    numScoops = 0
    scoops = k
    for i in range(g):
        useScoop = scoops % 2
        scoops //= 2
        if not useScoop:
            continue
        
        numScoops += 1
        for j in range(v):
            amts[j] += feed[i][j]
    
    for i in range(v):
        if amts[i] < vit[i]:
            break
    else:
        if numScoops < lo:
            lo = numScoops
            loScoops = k

ans = [str(lo)]
i = 1
while loScoops:
    if loScoops % 2:
        ans.append(str(i))
    i += 1
    loScoops //= 2

with open('holstein.out', 'w') as f:
    f.write(' '.join(ans) + '\n')